### PR TITLE
MSQ WorkerImpl: Ignore ServiceClosedException on postCounters.

### PR DESCRIPTION
**Pull request by gianm**
A race can happen where postCounters is in flight while the controller goes offline. When this happens, we should ignore the ServiceClosedException and continue without posting counters.

### CODE REVIEW

File: extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java [hunk 1]
```diff
@@ -850,7 +851,16 @@ private void postCountersToController() throws IOException
...
-      controllerClient.postCounters(id(), snapshotsTree);
+      try {
+        controllerClient.postCounters(id(), snapshotsTree);
+      } catch (IOException e) {
+        if (e.getCause() instanceof ServiceClosedException) {
+          // Suppress. This can happen if the controller goes away while a postCounters call is in flight.
+          log.debug(e, "Ignoring failure on postCounters, because controller has gone away.");
+        }
+
+        throw e;
```
Comment (georgew5656) [inline extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java:862]: won't this just rethrow the ServiceClosedException with a extra log?

**Review by LakshSingla — APPROVED** [review]

**Comment by gianm** [comment]
CI is failing due to code coverage. This particular section of code isn't super easy to write a test for. I could do it, but before tackling that I wanted to ask if people would be OK with merging this without coverage?

**Comment by georgew5656** [comment]
> CI is failing due to code coverage. This particular section of code isn't super easy to write a test for. I could do it, but before tackling that I wanted to ask if people would be OK with merging this without coverage?

works for me

**Review by cryptoe — APPROVED** [review]

**Comment by cryptoe** [comment]
Thanks for the PR @gianm!!

**Pull request merged**